redaxo/src/core/lib/util/editor.php

```diff
@@ -0,0 +1,54 @@
+<?php
+
+class rex_editor
+{
+    use rex_factory_trait;
+
+    private $editors = [
+        'emacs' => 'emacs://open?url=file://%file&line=%line',
+        'idea' => 'idea://open?file=%file&line=%line',
+        'macvim' => 'mvim://open/?url=file://%file&line=%line',
+        'phpstorm' => 'phpstorm://open?file=%file&line=%line',
+        'sublime' => 'subl://open?url=file://%file&line=%line',
+        'textmate' => 'txmt://open?url=file://%file&line=%line',
+        'vscode' => 'vscode://file/%file:%line',
+    ];
+
+    public function urlFromFile($filePath, $line)
+    {
+        // make internal urls work with parse_url()
+        $filePath = str_replace('rex:///', 'rex://', $filePath);
+
+        $parsedUrl = parse_url($filePath);
+        // rex:// internal url mapping to backend form-edit urls
+        if ($parsedUrl['scheme'] === 'rex') {
```

Comment:
ich hab das parsing von internen links wie rex://template/4 etc. erstmal hier hardcodiert.

wenn man das developer addon verwendet, würde man hier vermutlich eine source datei öffnen wollen, anstatt der backend webpages aufzurufen.

zunächst hatte ich hier versucht einen registrierungsmechanismus zu schaffen via EP o.ä., aber das geht nicht so wirklich wenn errors sehr früh auftreten.
man bräuchte hier einen mechanismus der direkt beim booten schon weiß wie die pfade in urls umgesetzt werden müssten, z.b. via `rex_config()` o.ä. - sodass die url umsetzungen auch bekannt sind wenn die addons noch nicht alle gebooted sind etc.

ideen dazu sind willkommen...

Reply:
Ich denke, ich würde das trotzdem über einen EP lösen. 
Wenn ein Template oder Modul ausgeführt wird, sind ja zum Beispiel die Addons schon geladen.

Kann da aber gerne auch dann einen nachgelagerten PR zu erstellen mit einem Vorschlag.